src/Site/SiteAwareTrait.php

```diff
@@ -28,96 +31,86 @@ public function setSites(Sites $sites)
     }
 
     /**
-     * @return Sites The sites collection for the authenticated user.
+     * @return Sites
+     *   The sites' collection for the authenticated user.
      */
     public function sites(): Sites
     {
         return $this->sites;
     }
 
     /**
-     * Look up a site by id.
+     * Returns the site by site UUID, site name or `site-name.env`.
      *
-     * @param string $site_id Either a site's UUID or its name
-     * @return null | Site
+     * @param string $site_id
+     *   Either a site's UUID or its name or site_env.
+     * @return Site
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
      */
-    public function getSite($site_id) : ?Site
+    public function getSite(string $site_id): Site
     {
-        if (strpos($site_id, ".")) {
-            $split = explode(".", $site_id);
-            $site_id = array_shift($split);
+        if (false !== strpos($site_id, '.')) {
+            $site_env_parts = explode('.', $site_id);
+            $site_id = $site_env_parts[0];
         }
+
         return $this->sites()->get($site_id);
     }
 
     /**
-     * Get the site and environment with the given ids, if provided
+     * Returns the environment by `site-name.env`.
+     *
+     * @param string $site_env
      *
-     * @param string $site_env_id The site/environment id in the form [<site>[.<env>]]
-     * @return array The site and environment in an array, if provided; may return [null, null]
+     * @return \Pantheon\Terminus\Models\Environment
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
+     * @throws \Pantheon\Terminus\Exceptions\TerminusNotFoundException
      */
-    public function getOptionalSiteEnv($site_env_id)
+    public function getEnv(string $site_env): TerminusModel
     {
-        if (empty($site_env_id)) {
-            return [null, null];
+        if (false === strpos($site_env, '.')) {
+            throw new TerminusException('The environment argument must be given as <site_name>.<environment>');
         }
 
-        list($site_id, $env_id) = array_pad(explode('.', $site_env_id), 2, null);
+        $site_env_parts = explode('.', $site_env);
+        $site_id = $site_env_parts[0];
+        $env_id = $site_env_parts[1];
 
-        $site = $this->getSite($site_id);
-        $env = !empty($env_id) ? $site->getEnvironments()->get($env_id) : null;
-        return [$site, $env];
+        return $this->sites()->get($site_id)->getEnvironments()->get($env_id);
```

Comment:
Are these methods going to throw exceptions equivalent to the ones removed from the callers if an environment does not exist?

Reply:
> Are these methods going to throw exceptions equivalent to the ones removed from the callers if an environment does not exist?

yes